codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/serialize/JsonSerializerGenerator.kt

```diff
@@ -55,10 +55,8 @@ import software.amazon.smithy.rust.codegen.util.outputShape
  * Class describing a JSON section that can be used in a customization.
  */
 sealed class JsonSection(name: String) : Section(name) {
-    /**
-     * Write the customization just before `object.finish()`.
-     */
-    data class FinalizeObject(val structureShape: StructureShape) : JsonSection("FinalizeObject")
+    /** Write the server error __type customization just before `object.finish()` */
```

Comment:
this customization is `__type` specific right, that's just an example? Maybe
```
Mutate the server error object prior to finalization. Eg: this can be used to inject `__type` to record the error type.

Use `jsonObject` to refer to the JSON builder.
```

Reply:
Yeah, you are right..